### PR TITLE
[#9790] Remove support for Python 2 in _setup.py

### PR DESCRIPTION
**Pull request by hawkowl**
https://twistedmatrix.com/trac/ticket/9790#ticket

This also removes some old OSes (solaris 10???) and bumps a few dependencies to be simpler (idna mostly). It also has the py2 removal changelog entry.

### CODE REVIEW

File: src/twisted/python/_setup.py [hunk 1]
```diff
@@ -81,34 +75,25 @@
...
-        'idna >= 0.6, != 2.3',
+        'idna >= 2.4',
     ],
     conch=[
         'pyasn1',
         'cryptography >= 2.6',
         'appdirs >= 1.4.0',
         'bcrypt >= 3.0.0',
     ],
-    soap=['soappy'],
```
Comment (glyph) [inline src/twisted/python/_setup.py:111]: lol `twisted.web.soap` still exists

File: src/twisted/python/_setup.py [hunk 2]
```diff
@@ -262,21 +240,16 @@ class my_build_ext(build_ext_twisted):
...
         "zope.interface >= 4.4.2",
         "constantly >= 15.1",
         "incremental >= 16.10.1",
         "Automat >= 0.3.0",
         "hyperlink >= 17.1.1",
-        # PyHamcrest 1.10.0 is Python 3 only, but lacks package metadata that
-        # says so. This condition can be dropped when Twisted drops support for
-        # Python 2.7.
-        "PyHamcrest >= 1.9.0, != 1.10.0",
+        "PyHamcrest >= 1.9.0",
```
Comment (glyph) [inline src/twisted/python/_setup.py:252]: Shouldn't this be a test-only dependency?

File: src/twisted/python/_setup.py (hunk 1 above)
Comment (glyph) [reply src/twisted/python/_setup.py:111]: (Given its apparent total lack of test coverage I'm suggesting we delete it; anyone who wanted this functionality on python 3 would have noticed it was missing some time ago.)

File: src/twisted/python/_setup.py [hunk 3]
```diff
@@ -190,11 +173,6 @@ def __init__(self, *args, **kwargs):
             ],
         libraries=["ws2_32"],
         condition=lambda _: _isCPython and sys.platform == "win32"),
-
-    ConditionalExtension(
-        "twisted.python._sendmsg",
```
Comment (glyph) [inline src/twisted/python/_setup.py:195]: and good riddance!!!!!

**Review by rodrigc — APPROVED** [review]
This looks OK

**Pull request merged**